source/common/secret/secret_manager_impl.h

```diff
@@ -3,20 +3,35 @@
 #include <unordered_map>
 
 #include "envoy/secret/secret_manager.h"
+#include "envoy/server/instance.h"
 #include "envoy/ssl/tls_certificate_config.h"
 
 #include "common/common/logger.h"
+#include "common/secret/sds_api.h"
 
 namespace Envoy {
 namespace Secret {
 
 class SecretManagerImpl : public SecretManager, Logger::Loggable<Logger::Id::upstream> {
 public:
-  void addOrUpdateSecret(const envoy::api::v2::auth::Secret& secret) override;
-  const Ssl::TlsCertificateConfig* findTlsCertificate(const std::string& name) const override;
+  SecretManagerImpl(Server::Instance& server) : server_(server) {}
+
+  void addStaticSecret(const envoy::api::v2::auth::Secret& secret) override;
+  const Ssl::TlsCertificateConfig* findStaticTlsCertificate(const std::string& name) const override;
+
+  DynamicTlsCertificateSecretProviderSharedPtr findOrCreateDynamicTlsCertificateSecretProvider(
+      const envoy::api::v2::core::ConfigSource& config_source, const std::string& config_name,
+      Init::Manager& init_manager) override;
 
 private:
-  std::unordered_map<std::string, Ssl::TlsCertificateConfigPtr> tls_certificate_secrets_;
+  Server::Instance& server_;
```

Comment:
not used, remove this and header includes and bazel path includes if applicable.

Reply:
It is used. The server_ is passed to SdsApi constructor via findOrCreateDynamicTlsCertificateSecretProvider(). SdsApi::initialize() makes a call to Envoy::Config::SubscriptionFactory::subscriptionFromConfigSource() which needs information from server.

Reply:
OK